specifyweb/middleware/general.py

```diff
@@ -1,26 +1,46 @@
 import traceback
 import json
-from typing import Any, Optional, Union, Dict, FrozenSet, List
+from typing import Any, Optional, Union, Dict, FrozenSet, List, Callable, Literal
 from django import http
 from django.core import serializers
 from django.db import models
 from django.conf import settings
 from django.forms.models import model_to_dict
+from django.views.decorators.http import require_http_methods as raw_require_http_methods
+
+REQUEST_METHOD = Literal['HEAD', 'GET', 'POST', 'PUT', 'DELETE', 'OPTIONS']
+
+
+def require_GET(function: Callable):
+    return raw_require_http_methods(['HEAD', 'GET'])(function)
+
+
+def require_http_methods(_request_method_list: List[REQUEST_METHOD]):
+    request_method_list = set(_request_method_list)
+    if 'GET' in _request_method_list:
+        request_method_list.add('HEAD')
+
+    def callable(function: Callable):
+        return raw_require_http_methods(list(request_method_list))(function)
+        
+    return callable
```

Comment:
Realized that this can be simplified more.

```suggestion
    return raw_require_http_methods(list(request_method_list))
```

Reply:
fixed in https://github.com/specify/specify7/pull/4435/commits/e7843212e832270c0d5ef0b47356a08e09413cc4